### PR TITLE
Qt: Make Log and Log Configuration widget separate from dock

### PR DESCRIPTION
**Pull request by spycrab**
Needed before being able to properly work on the Debugger widgets

### CODE REVIEW

**Comment by spycrab** [comment]
@ligfx

File: Source/Core/DolphinQt2/Config/DockWidget.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  if (Settings::Instance().IsLogVisible())
+    m_tab_widget->addTab(m_log_widget, tr("Log"));
+
+  if (Settings::Instance().IsLogConfigVisible())
+    m_tab_widget->addTab(m_log_config_widget, tr("Log Configuration"));
+
+  setHidden(m_tab_widget->count() == 0);
+}
+
+bool DockWidget::eventFilter(QObject* object, QEvent* event)
```
Comment (ligfx) [inline Source/Core/DolphinQt2/Config/DockWidget.cpp]: This `eventFilter` function never gets called. Did you mean to override `event(QEvent*)`?

**Comment by spycrab** [comment]
@ligfx They're both QDockWidgets now. The layout issue should be fixed too.

**Comment by ligfx** [comment]
Now the tabs are on the bottom, for some reason:

screen shot 2017-09-14 at 10 45 27 am


Also, I can't get the widgets to tabify when they're floating, only when they're docked. Is that the desired behavior?

File: Source/Core/DolphinQt2/MainWindow.h [hunk 2]
```diff
@@ -17,8 +17,9 @@
 #include "DolphinQt2/ToolBar.h"
 
 struct BootParameters;
+class LogWidget;
```
Comment (ligfx) [inline Source/Core/DolphinQt2/MainWindow.h]: Alphabetically sort please

**Pull request merged**